### ISSUE TITLE
Launching the app VDMX disables AtemOSC / Osculator until restart

### ISSUE THREAD

**Issue by cybrgloom** [SteffeyDev/atemOSC#219]
Hello! When I open the app VDMX on ANY computer on my network, AtemOSC instantly and consistently stops working, until I quit and restart AtemOSC. Upon restarting AtemOSC, it continues to work exactly like it was before launching VDMX. 

I'm not sure where it exactly breaks, but VDMX, Osculator, Live seems to work like they always have, and I can see them being responsive to all messages, it is in AtemOSC messages stop appearing. So I'm reporting it here in hope that either it's a bug or I'm doing something wrong and can be adviced. I have turned on debug, but nothing appears in the log when this happens. It just stops working. Confirmed on multiple computers and operating systems. 

**How to reproduce:**
Download VDMX demo from Vidvox (https://vidvox.net)
Launch AtemOSC, connect to Atem Mini
Launch Osculator, see attached simple setup (routing midi notes to programs)
Launch any MIDI host, I'm using Live 11 for this test 
Verify that sending midi notes 0, 1, 2, 3 to Osculator instantly cuts on the Atem
Now launch VDMX. 
Observe within a second, AtemOSC stops responding. 
Quit and restart AtemOSC. 
Observe everything working like before.
Quit VDMX. Still works. 
Now Restart VDMX. Stops working... 

It is not only on localhost, I can launch VDMX on any computer on my network, the same always happens. AtemOSC stops responding on any computer on the same network. I have tried my setups on multiple computers, the problem happens every time everywhere. It is quite bewildering, I can't really fathom why this could be! 

I have no idea if this is a bug, or something broken on my system(s), or operator error, hopefully the steps above could indicate where the problem is? Let me know if I shall follow up with anything I can details/logs/tests. 

Tested on Macbook Pro 15" 2013, a 15" from 2017, and a 13" from 2019, with two different Atem Minis, AtemOSC/VDMX/Osculator/Live/macOS all on latest versions as of May 2nd 2021. Network is cabled, but this also happens on wireless. The problem was the same last summer, then I was on Mojave, tried and gave up on AtemOSC because the exact same thing happened. Now I see development has become active, and I'm on Catalina so I was trying again to see if AtemOSC can be used together with VDMX.

Cheers

![Screenshot 2021-05-02 at 17 26 20](https://user-images.githubusercontent.com/76553242/116818684-f75fcb00-ab6c-11eb-8744-dfe8e01f820f.png)

![Screenshot 2021-05-02 at 17 27 49](https://user-images.githubusercontent.com/76553242/116818712-047cba00-ab6d-11eb-9a2a-542ad7ed8f3b.png)

**Comment by randallpacker** [comment]
Hey Cybergloom... I use VDMX to control atemOSC, but I don't use anything in between. In other words, you can send OSC directly from VDMX. There are VDMX tutorials on how to do that, unless it is something you have tried already. Good luck!

**Comment by cybrgloom** [comment]
Hey thanks! My challenge is that the "brain" of my setup is one computer running Ableton Live, which then via MIDI controls all further cameras, lights, devices (Atems for example) and then multiple computers running VDMX for the visuals. The VDMX computers are "passive", they only get trigger messages.   

However, this does inform me since you obviously do not have any problems, that when you launch VDMX, your atemOSC keeps working? I do have a suspicion that something is broken in either my setup or my message flow (this COULD be operator error), since this is such a weird issue, it would have been noticed and reported by many by now.

**Comment by xenonheavyindustries** [comment]
Have you done any tracing/debugging with Protokol and/or Midi Monitor? That should help determine where the issue lies.

**Comment by randallpacker** [comment]
I would create a simple patch in VDMX and send OSC directly to atemOSC to see if you have the same problem. I have a similar setup, using Ableton Live with VDMX. You can also use the OSCGrabber Max4Live to control atemOSC, or, you can send MIDI to VDMX, and then when it is received, it in turns sends OSC to atemOSC. I use both methods depending on what I am trying to do.

**Comment by cybrgloom** [comment]
Thank you both, good suggestions, I now tested with a simple LiveGrabber MaxForLive device directly sending the OSC messages from Live to AtemOSC, without Osculator. That works great, and with this setup, launching VDMX works fine so far. 

So my problem is now better specificed, when launching VDMX, the connection Osculator -> AtemOSC is severed. Monitoring signals I can see they run into Osculator and Osculator outputs them but they do not reach AtemOSC once VDMX is launched. 

But, I now have a neat workaround that is actually an improvement, building a direct OSC solution within Live seems even better for me (as long as it is stable) so I'll just keep using this for now and ignore the Osculator -> AtemOSC route. 

Thank you for helping me!

**Comment by randallpacker** [comment]
Great! The Max4Live LiveGrabber is very reliable, I have never had a problem with it. This is the first time someone has asked a question about Ableton & VDMX, so I had to answer... best of luck! Randall

**Comment by SteffeyDev** [comment]
I only followed about half of that, but sounds like you guys had a productive discussion. Am I good to close this? I don't think there's much I can do about the original issue, if AtemOSC never gets the packets then is beyond my control. Might be an issue with the OSC library I am using, but that's not something I have time to dig into right now.

**Comment by cybrgloom** [comment]
Yeah you can close it, thanks for following up. I have a workaround (that
improves my setup, so win-win). if I run into the issue again in another
context I will ask in a new post, hopefully with more details as to
why/where it happens. Cheers!
